libs/ledger/ledger/injections.py

```diff
@@ -46,8 +69,63 @@ def mass_2_source(self):
         return self.mass_2 / (1 + self.redshift)
 
     @property
-    def chirp_mass(self):
-        return chirp_mass(self.mass_1, self.mass_2)
+    def luminosity_distance(self, cosmology=DEFAULT_COSMOLOGY):
+        return cosmology.luminosity_distance(self.redshift).value
+
+
+@dataclass
+class PycbcParameterSet(ExtrinsicParameterSet, IntrinsicParameterSet):
+    def convert_to_lal_params(self, reference_frequency: float):
+        self.inclination = np.zeros(len(self))
+        self.spin1x = np.zeros(len(self))
+        self.spin1y = np.zeros(len(self))
+        self.spin1z = np.zeros(len(self))
+        self.spin2x = np.zeros(len(self))
+        self.spin2y = np.zeros(len(self))
+        self.spin2z = np.zeros(len(self))
+
+        for i in range(len(self)):
+            (
+                self.inclination[i],
+                self.spin1x[i],
+                self.spin1y[i],
+                self.spin1z[i],
+                self.spin2x[i],
+                self.spin2y[i],
+                self.spin2z[i],
+            ) = bilby_to_lalsimulation_spins(
+                a_1=self.a_1[i],
+                a_2=self.a_2[i],
+                tilt_1=self.tilt_1[i],
+                tilt_2=self.tilt_2[i],
+                phi_12=self.phi_12[i],
+                phi_jl=self.phi_jl[i],
+                mass_1=self.mass_1[i],
+                mass_2=self.mass_2[i],
+                theta_jn=self.theta_jn[i],
+                phase=self.phase[i],
+                reference_frequency=reference_frequency,
+            )
+
+    def waveform_generation_params(self, reference_frequency: float):
+        # For clarity, explicitly define the parameter dictionary
+        # needed for waveform generation
+        self.convert_to_lal_params(reference_frequency)
```

Comment:
This is a little confusing b/c the `PycbcParameterSet` actually instantiates with `bilby` params and does the conversion under the hood. 

Would it make more sense to add an additional `BilbyParameterSet` class and have methods for each that converts between the two? 

(in that case, I think re-naming the `PycbcParameterSet` --> `LALParameterSet` might make more sense too.

Reply:
If it's too difficult can just implement the `BilbyParameterSet` --> `LALParameterSet` conversion for now and leave the reverse direction (which i doubt we'll really use) for later